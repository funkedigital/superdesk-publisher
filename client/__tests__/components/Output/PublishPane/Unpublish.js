import React from "react";
import Unpublish from "../../../../components/Output/PublishPane/Unpublish";
import { render, wait } from "@testing-library/react";
import Store from "../../../../components/Output/Store";

const destinations = [
  { status: "published", tenant: { code: 'tenant1', name: 'tenant1' } },
  { status: "published", tenant: { code: 'tenant2', name: 'tenant2' } }
];

describe("Output/PublishPane/Unpublish", () => {
  it("renders properly", async () => {
    const { container, getByText } = render(
      <Store.Provider
        value={{
          publisher: {},
          selectedItem: { id: 1 }
        }}
      >
        <Unpublish
          destinations={destinations}
        />
      </Store.Provider>
    );

    await wait(() =>
      expect(container.querySelector(".sd-loader")).not.toBeInTheDocument(),
    )

    expect(container.firstChild).toMatchSnapshot();
  });
});
