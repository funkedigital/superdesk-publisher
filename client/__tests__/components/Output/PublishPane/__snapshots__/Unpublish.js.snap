// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/PublishPane/Unpublish renders properly 1`] = `
<div
  class="side-panel__content"
>
  <div
    class="side-panel__content-block side-panel__content-block--pad-small"
  >
    <div
      class="side-panel__content-block-heading side-panel__content-block-heading--small-margin"
    />
  </div>
  <div
    class="side-panel__content-block"
  >
    <div
      class="form__row form__row--small-padding"
    >
      <div
        class="flex-grid flex-grid--boxed-small flex-grid--wrap-items flex-grid--small-1"
      >
        <button
          class="btn__check flex-grid__item"
        >
          tenant1
        </button>
        <button
          class="btn__check flex-grid__item"
        >
          tenant2
        </button>
      </div>
    </div>
  </div>
</div>
`;
